src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -0,0 +1,184 @@
+        }
+
+        private Glyph GetGlyph(SymbolKind kind, Accessibility declaredAccessibility)
+        {
+            Glyph publicIcon;
+            switch (kind)
+            {
+                case SymbolKind.Field:
+                    publicIcon = Glyph.FieldPublic;
+                    break;
+                case SymbolKind.Local:
+                    publicIcon = Glyph.Local;
+                    break;
+                case SymbolKind.Method:
+                    publicIcon = Glyph.MethodPublic;
+                    break;
+                case SymbolKind.Parameter:
+                    publicIcon = Glyph.Parameter;
+                    break;
+                case SymbolKind.Property:
+                    publicIcon = Glyph.PropertyPublic;
+                    break;
+                case SymbolKind.RangeVariable:
+                    publicIcon = Glyph.RangeVariable;
+                    break;
+                case SymbolKind.TypeParameter:
+                    publicIcon = Glyph.TypeParameter;
+                    break;
+                default:
+                    throw new ArgumentException();
+            }
+
+            switch (declaredAccessibility)
+            {
+                case Accessibility.Private:
+                    publicIcon += Glyph.ClassPrivate - Glyph.ClassPublic;
+                    break;
+
+                case Accessibility.Protected:
+                case Accessibility.ProtectedAndInternal:
+                case Accessibility.ProtectedOrInternal:
+                    publicIcon += Glyph.ClassProtected - Glyph.ClassPublic;
+                    break;
+
+                case Accessibility.Internal:
+                    publicIcon += Glyph.ClassInternal - Glyph.ClassPublic;
+                    break;
+            }
+
+            return publicIcon;
+        }
+
+        private void AddBuilder(CompletionContext completionContext)
+        {
+            completionContext.SuggestionModeItem = CommonCompletionItem.Create(CSharpFeaturesResources.Name, CompletionItemRules.Default);
+        }
+
+        private ITypeSymbol UnwrapType(ITypeSymbol type)
+        {
+            var nts = type as INamedTypeSymbol;
+            if (nts != null && nts.ConstructedFrom != null)
+            {
+                var constructedFrom = nts.ConstructedFrom;
+                switch (constructedFrom.SpecialType)
+                {
+                    case SpecialType.System_Collections_Generic_IEnumerable_T:
+                    case SpecialType.System_Collections_Generic_IList_T:
+                    case SpecialType.System_Collections_Generic_ICollection_T:
+                    case SpecialType.System_Collections_Generic_IReadOnlyList_T:
+                    case SpecialType.System_Collections_Generic_IReadOnlyCollection_T:
+                    case SpecialType.System_Nullable_T:
+                        return UnwrapType(nts.TypeArguments[0]);
+                }
+
+                if (constructedFrom.Name == "Task"
```

Comment:
you can just get the ITypeSymbols for ```Task/Task<T>/ValueTask<T>``` off the compilatoin. I recently wrote this in another PR.  Just extract that into a useful space and test if your ConstructedFRom is equal to any of those.